widget/map.go

```diff
@@ -29,15 +29,76 @@ type Map struct {
 	zoom, x, y int
 
 	cl *http.Client
+
+	tileSource      string // url to download xyz tiles (example: "https://tile.openstreetmap.org/%d/%d/%d.png")
+	hideDisclaimer  bool   // enable copyright disclaimer
+	disclaimerLabel string // label for disclaimer (example: "OpenStreetMap")
+	disclaimerUrl   string // url for disclaimer (example: "https://openstreetmap.org")
+	hideZoomButtons bool   // enable zoom buttons
+	hideMoveButtons bool   // enable move map buttons
+}
+
+type MapOption func(*Map)
+
+func WithOsmTiles() MapOption {
+	return func(m *Map) {
+		m.tileSource = "https://tile.openstreetmap.org/%d/%d/%d.png"
+		m.disclaimerLabel = "OpenStreetMap"
+		m.disclaimerUrl = "https://openstreetmap.org"
+		m.hideDisclaimer = false
+	}
+}
+
+func WithTileSource(tileSource string) MapOption {
+	return func(m *Map) {
+		m.tileSource = tileSource
+	}
+}
+
+func WithDisclaimer(enable bool, label, url string) MapOption {
+	return func(m *Map) {
+		m.hideDisclaimer = !enable
+		m.disclaimerLabel = label
+		m.disclaimerUrl = url
+	}
+}
+
+func WithZoomButtons(enable bool) MapOption {
+	return func(m *Map) {
+		m.hideZoomButtons = !enable
+	}
+}
+
+func WithScrollButtons(enable bool) MapOption {
+	return func(m *Map) {
+		m.hideMoveButtons = !enable
+	}
 }
 
 // NewMap creates a new instance of the map widget.
 func NewMap() *Map {
-	m := &Map{cl: &http.Client{}}
+	m := &Map{
+		cl:              &http.Client{},
+		tileSource:      "https://tile.openstreetmap.org/%d/%d/%d.png",
```

Comment:
Can we avoid this duplication of important strings by delegating to the `Options` code you added perhaps?

Reply:
Good idea. I updated NewMap() with your suggestion.